client/state/sites/reducer.js

```diff
@@ -204,6 +205,17 @@ export function items( state = {}, action ) {
 			}, state );
 		}
 
+		case SITE_DOMAINS_PRIMARY_SET_COMPLETED: {
+			const { siteId, primaryDomainName } = action;
+			return {
+				...state,
+				[ siteId ]: {
+					...state[ siteId ],
+					URL: `http://${ primaryDomainName }`
```

Comment:
I figured out it's too complicated to update the existing endpoint, but I found another simple solution :)